File: src/main/kotlin/com/miro/miroappoauth/dto/ShapeType.java
package com.miro.miroappoauth.dto;

import org.jetbrains.annotations.Nullable;

import java.util.stream.Stream;

public enum ShapeType {
    rectangle,
    circle;

    public static ShapeType fromString(@Nullable String value) {
        return Stream.of(values())
                .filter(cnst -> cnst.name().equalsIgnoreCase(value))
                .findFirst()
                .orElse(rectangle);
    }
}

File: src/main/kotlin/com/miro/miroappoauth/client/v2/ShapeType.kt
package com.miro.miroappoauth.client.v2

import java.util.stream.Stream

enum class ShapeType {
    rectangle,
    triangle,
    circle;

    companion object {
        fun fromString(value: String?): ShapeType {
            return Stream.of(*values())
                .filter { cnst: ShapeType -> cnst.name.equals(value, ignoreCase = true) }
                .findFirst()
                .orElse(rectangle)
        }
    }
}
